nautobot/dcim/forms.py

```diff
@@ -2196,6 +2197,43 @@ def __init__(self, *args, **kwargs):
         self.fields["untagged_vlan"].widget.add_query_param("location", device.location.pk)
         self.fields["tagged_vlans"].widget.add_query_param("location", device.location.pk)
 
+    def clean(self):
+        super().clean()
+        ip_addresses = self.cleaned_data.get("ip_addresses")
+        # If there is no ip_addresses field here
+        # we know it is a CreateForm
+        if ip_addresses is None:
+            pass
```

Comment:
I should be checking if the `interface`/`vm_interface` exist in the database yet instead of checking `ip_addresses` here to determine whether if it is a `CreateForm` or an `EditForm`. But it will be an empty list if you remove all IPs.